rollout/trafficrouting/istio/istio.go

```diff
@@ -780,18 +804,18 @@ func (r *Reconciler) generateHeaderBasedPatches(httpRoutes []VirtualServiceHTTPR
 	return patches
 }
 
-func hasHeaderRoute(httpRoutes []VirtualServiceHTTPRoute) bool {
-	for _, route := range httpRoutes {
-		if route.Name == HeaderRouteName {
-			return true
+func hasHeaderRoute(headerRouting *v1alpha1.SetHeaderRoute, httpRoutes []VirtualServiceHTTPRoute) (bool, int) {
```

Comment:
As we providing an index in response I think no need to have a `bool` flag anymore. We can use -1 as not found.
And rename it to `headerRouteIndex` or so.